### DISCUSSION TITLE
LightningCLI: Set interval for lr_scheduler in config.yaml

### DISCUSSION THREAD

**Discussion by nicoloesch** [Lightning-AI/pytorch-lightning#17868] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi everyone,

I recently changed over to `LightningCLI` from the regular `argparser`. In order to change `lr_scheduler` and `optimizer` from the command-line, I added the respective keys to the `config.yaml`
```yaml
optimizer:
  class_path: torch.optim.Adam
  init_args:
    lr: 1e-5
    weight_decay: 0.0
lr_scheduler:
  class_path: adiff.utils.callbacks.WarmUpScheduler
  init_args:
    warmup: 20
```

However, `.step()` of the `lr_scheduler` is only called every epoch. **I want it to be called every step**. 

In `configure_optimizers` of a `LightningModule`, one can set the `interval` of the scheduler. Is this also possible through the `CLI` interface/ in the `yaml`-config and if so, how?  I tried multiple different versions but none of them was parse-able.

Thank you for any help!
Nico

**Comment by mauvilsa** [comment, marked as answer]
See https://github.com/Lightning-AI/lightning/issues/15340#issuecomment-1434560283
